wordpress-mobile/WordPress-iOS
# Replace `BlogJetpackSettingsService.context` with `CoreDataStack`

Currently the Jetpack settings are modified in view controllers (i.e. [this one here](https://github.com/wordpress-mobile/WordPress-iOS/blob/595e5e1d335cf6c27bddc0d385637a1e1efb7bbf/WordPress/Classes/ViewRelated/Jetpack/Jetpack%20Settings/JetpackSettingsViewController.swift#L192)) and saved when API call succeed (i.e. [this code here](https://github.com/wordpress-mobile/WordPress-iOS/blob/595e5e1d335cf6c27bddc0d385637a1e1efb7bbf/WordPress/Classes/Services/BlogJetpackSettingsService.swift#L116-L121)).

This PR refactors the saving Jetpack settings part, by using `performAndSave`, so that we don't write in the main context.

## Test Instructions

Go to a Jetpack connected test site from the app, navigate to "My Sites" -> "Menu" -> "Jetpack Settings", verify settings can be changed and saved in the Jetpack Settings screen.

## Regression Notes
1. Potential unintended areas of impact
None.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
None.

3. What automated tests I added (or what prevented me from doing so)
None.

PR submission checklist:
